### PR TITLE
Add Sendmail integration

### PR DESCRIPTION
**Pull request by dabcoder**
### What does this PR do?

Adds a sendmail integration to monitor mail queues.

### Motivation

Customer request.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

File: sendmail/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+### Events
+
+Sendmail does not include any events.
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][5].
+
+[1]: https://www.proofpoint.com/us/open-source-email-solution
+[2]: https://github.com/DataDog/integrations-extras/blob/master/sendmail/datadog_checks/sendmail/data/conf.yaml.example
```
Comment (ofek) [inline sendmail/README.md]: Wrong link

File: sendmail/datadog_checks/sendmail/sendmail.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        # mailq sample output
+        """
+        MSP Queue status...
+        /var/spool/mqueue-client is empty
+            Total requests: 0
+        MTA Queue status...
+        /var/spool/mqueue is empty
+            Total requests: 0
+        """
```
Comment (ofek) [inline sendmail/datadog_checks/sendmail/sendmail.py]: Let's use a comment (`#`) instead

File: sendmail/datadog_checks/sendmail/sendmail.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        # if we want to use sendmail, we need to append -bp to it
+        # https://www.electrictoolbox.com/show-sendmail-mail-queue/
+        if "sendmail" in sendmail_command:
+            command = [sendmail_command, '-bp']
+        else:
+            command = [sendmail_command]
+
+        # Listing the directory might require sudo privileges
+        if use_sudo:
+            test_sudo = os.system('setsid sudo -l < /dev/null')
```
Comment (ofek) [inline sendmail/datadog_checks/sendmail/sendmail.py]: Let's just try it instead like https://github.com/DataDog/integrations-core/blob/712bbee9135f5ac0a05f830d86f188cadb287ee1/process/datadog_checks/process/process.py#L201

File: sendmail/datadog_checks/sendmail/sendmail.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+from datadog_checks.base import AgentCheck, ConfigurationError, is_affirmative
+from datadog_checks.utils.subprocess_output import get_subprocess_output
+
+
+class SendmailCheck(AgentCheck):
+
+    SERVICE_CHECK_NAME = 'sendmail.returns.output'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
```
[the remark this answers is not in the document]
Comment (dabcoder) [reply sendmail/datadog_checks/sendmail/sendmail.py]: Should I remove the whole `__init__` method?

[the remark this answers is not in the document]
Comment (ofek) [reply sendmail/datadog_checks/sendmail/sendmail.py]: Yup!

**Pull request merged**